bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java

```diff
@@ -115,15 +116,18 @@ private void cleanup(final ResourceResolver resolver, final int batchSize, final
         long total = 0;
         boolean retried = false;
         SocialResourceConfiguration config = socialUtils.getDefaultStorageConfig();
-        SocialResource rootResource = (SocialResource) resolver.getResource(config.getAsiPath());
-        rootResource.getResourceProvider().setConfig(config);
-        rootResource.getResourceProvider().delete(resolver, path);
-        try {
-            resolver.commit();
-        } catch (PersistenceException e) {
-            LOG.info("Could not persist deletion. Maybe there is nothing to delete", e);
-        }
-
+        // For SRPs that handle recursive delete
+        if (config.getAsiPath().equals(JCR_ASI_PATH)) {
```

Comment:
Should it be equals() or startsWith()?

Reply:
It is exact ASI path to be compared so used equals(). startWith() will also work for the same.
Should it be modified?